src/app/clusters/channel-server/channel-server.cpp

```diff
@@ -162,15 +162,15 @@ CHIP_ERROR ChannelAttrAccess::Read(const app::ConcreteReadAttributePath & aPath,
     switch (aPath.mAttributeId)
     {
     case app::Clusters::Channel::Attributes::ChannelList::Id: {
-        if (isDelegateNull(delegate, endpoint))
+        if (isDelegateNull(delegate, endpoint) && !HasFeature(endpoint, ChannelFeature::kChannelList))
```

Comment:
The conditional operator was wrong. I've updated it to OR. 
So in case we don't have delegate or endpoint doesn't have a feature, clients will receive an error or empty list as a response to read command.